tests/100-gotest.sh

```diff
@@ -1,4 +1,13 @@
 #!/bin/sh
 
 GOFILES="$(go list ./... | grep -v vendor)"
-exec go test ${GOFILES}
+
+failed=0
+for gofile in $GOFILES; do
+	go test "$gofile"
```

Comment:
This actually has an unpleasant side effect:
If you do a single call like previously
```
go test $(go list ./... | grep -v vendor)
```
then you get compilation once. Of you do `go test` for each directory separately, you get compilation each time, and the run time is roughly multiplied by 10 or so, which is quite nasty, because each to test run takes roughly 1m20s to 1m40 on my machine.

Reply:
OK, that's a very good point. I did this in preparation to use the -coverprofile option to go test. If -coverprofile is used when specifying multiple packages it refuses to run reporting, "cannot use test profile flag with multiple packages". What we can do instead is a top-level conditional that runs the multiple packages as one command iff coverage is not requested. When coverage is requested we'll run go test in a loop this way. Sound reasonable?